system-variables.md

```diff
@@ -1,129 +1,836 @@
+- When the parameter of the aggregate function is not distinct, `HashAgg` is run concurrently and respectively in two phases - the `partial` phase and the `final` phase.
+
+### tidb_hashagg_partial_concurrency
+
+- Scope: SESSION | GLOBAL
+- Default value: 4
+- This variable is used to set the concurrency of executing the concurrent `hash aggregation` algorithm in the `partial` phase.
+- When the parameter of the aggregate function is not distinct, `HashAgg` is run concurrently and respectively in two phases - the `partial` phase and the `final` phase.
+
+### tidb_index_join_batch_size
+
+- Scope: SESSION | GLOBAL
+- Default value: 25000
+- This variable is used to set the batch size of the `index lookup join` operation.
+- Use a bigger value in OLAP scenarios, and a smaller value in OLTP scenarios.
+
+### tidb_index_lookup_concurrency
+
+- Scope: SESSION | GLOBAL
+- Default value: 4
+- This variable is used to set the concurrency of the `index lookup` operation.
+- Use a bigger value in OLAP scenarios, and a smaller value in OLTP scenarios.
+
+### tidb_index_lookup_join_concurrency
+
+- Scope: SESSION | GLOBAL
+- Default value: 4
+- This variable is used to set the concurrency of the `index lookup join` algorithm.
+
+### tidb_index_lookup_size
+
+- Scope: SESSION | GLOBAL
+- Default value: 20000
+- This variable is used to set the batch size of the `index lookup` operation.
+- Use a bigger value in OLAP scenarios, and a smaller value in OLTP scenarios.
+
+### tidb_index_serial_scan_concurrency
 
-### The working mechanism of system variables
+- Scope: SESSION | GLOBAL
+- Default value: 1
+- This variable is used to set the concurrency of the `serial scan` operation.
+- Use a bigger value in OLAP scenarios, and a smaller value in OLTP scenarios.
 
-* Session variables will only initialize their own values based on global variables when a session is created. Changing a global variable does not change the value of the system variable being used by the session that has already been created.
+### tidb_init_chunk_size
+
+- Scope: SESSION | GLOBAL
+- Default value: 32
+- This variable is used to set the number of rows for the initial chunk during the execution process.
+
+### tidb_isolation_read_engines <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION
+- Default value: tikv, tiflash, tidb
+- This variable is used to set the storage engine list that TiDB can use when reading data.
+
+### tidb_low_resolution_tso
+
+- Scope: SESSION
+- Default value: 0
+- This variable is used to set whether to enable the low precision TSO feature. After this feature is enabled, new transactions use a timestamp updated every 2 seconds to read data.
+- The main applicable scenario is to reduce the overhead of acquiring TSO for small read-only transactions when reading old data is acceptable.
+
+### tidb_max_chunk_size
+
+- Scope: SESSION | GLOBAL
+- Default value: 1024
```

Comment:
the minimum value is 32. It's not recommended to set it too large for cache locality reasons.